src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2637,6 +2637,12 @@ private TypeCompareState compareTypesForEquality(CORINFO_CLASS_STRUCT_* cls1, CO
             // If neither type is a canonical subtype, type handle comparison suffices
             if (!type1.IsCanonicalSubtype(CanonicalFormKind.Any) && !type2.IsCanonicalSubtype(CanonicalFormKind.Any))
             {
+                if (!exact)
+                {
+                    // Compare with Enum's underlying primitive type
+                    type1 = type1.IsEnum ? type1.UnderlyingType : type1;
+                    type2 = type2.IsEnum ? type2.UnderlyingType : type2;
```

Comment:
`UnderlyingType` returns `this` if the type is not an enum.

```suggestion
                    type1 = type1.UnderlyingType;
                    type2 = type2.UnderlyingType;
```